src/types/UiaTextRangeBase.cpp

```diff
@@ -1052,14 +1025,8 @@ try
     constexpr auto endpoint = TextPatternRangeEndpoint::TextPatternRangeEndpoint_Start;
     const auto bufferSize{ _pData->GetTextBuffer().GetSize() };
     const auto documentEnd = _getDocumentEnd();
-    if (bufferSize.CompareInBounds(_start, documentEnd, true) > 0)
-    {
-        _start = documentEnd;
-    }
-    if (bufferSize.CompareInBounds(_end, documentEnd, true) > 0)
-    {
-        _end = documentEnd;
-    }
+    _start = std::min(_start, documentEnd);
+    _end = std::min(_end, documentEnd);

```

Comment:
I'm conflicted. On one hand, `IsDegenerate()` would be fixed, but on the other hand, we clearly messed up if `_start > _end` at any point. These two lines help alleviate that by clamping down to `documentEnd`. Maybe adding an `assert(_start <= _end)` in `IsDegenerate()` would be good to detect if we ever run into that? idk, if that would be useful though. idk.

Reply:
I'm not sure what to do either, but I personally would make the change to `IsDegenerate()` regardless. Simply because bugs will exist in our code regardless of how hard we try, so we might as well build our code around that assumption. But I could also be completely wrong about it here, since I'm not completely familiar with how `IsDegenerate()` is used and whether such a change would make any tangible safety improvements at all in the first place.
`assert()`s  are IMO a great idea in either case, and the more we use them the better, because it's like test driven programming, but at runtime and it's more productive (arguably). 😅